### PR TITLE
mds: fix scan_stray_dir not reset next.frag on each run of stray inode

### PR DESCRIPTION
**Pull request by ethanwu-syno**
When we finish scanning one stray inode we should reset next.frag so that next stray inode could start from its first dir fragment rather than last stray inode's last frag.

Fixes: https://tracker.ceph.com/issues/58294
Signed-off-by: ethanwu <ethanwu@synology.com>

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "pacific"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

## Checklist
- Tracker (select at least one)
  - [x] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [x] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

File: src/mds/MDCache.cc [hunk 1]
```diff
@@ -10120,6 +10120,7 @@ void MDCache::scan_stray_dir(dirfrag_t next)
 	}
       }
     }
+    next.frag = frag_t();
```
Comment (vshankar) [inline src/mds/MDCache.cc:10123]: Looks correct. You found this bug as not all stray inodes were getting evaluated, yes?

**Comment by vshankar** [comment]
Also requires a tracker since we would want to backport this change.

File: src/mds/MDCache.cc (hunk 1 above)
Comment (ethanwu-syno) [reply src/mds/MDCache.cc:10123]: Yes, I did a test that would produce a lot of stray dentries and found out that many of them didn't get evaluated after MDS restarted.

**Comment by ethanwu-syno** [comment]
> Also requires a tracker since we would want to backport this change.

OK, I'll create a ticket under
https://tracker.ceph.com/projects/cephfs

**Comment by vshankar** [comment]
> > Also requires a tracker since we would want to backport this change.
> 
> OK, I'll create a ticket under https://tracker.ceph.com/projects/cephfs

... and include the tracker link in the commit message as `Fixes: ...` tag.

**Comment by vshankar** [comment]
jenkins test make check

**Comment by vshankar** [comment]
jenkins test make check

**Review by vshankar — APPROVED** [review]
https://tracker.ceph.com/projects/cephfs/wiki/Main#15-Dec-2022

**Comment by vshankar** [comment]
jenkins retest make check

**Comment by vshankar** [comment]
jenkins test make check

**Comment by vshankar** [comment]
jenkins test make check

**Pull request merged**